release.py

```diff
@@ -288,11 +288,7 @@ def format_changes(categories: dict[str, list[tuple[str, str]]], tag: str) -> st
         if items:
             lines.append(f'## {commit_type_to_category(commit_type)}\n')
             for description, pr_link in items:
-                pr_num = '?'
-                match = re.match(r'https?://[^ ]+/pull/(\d+)', pr_link)
-                if match:
-                    pr_num = match.group(1)
-                lines.append(f'* {description} (#{pr_num})')
+                lines.append(f'* {description} in {pr_link}')
```

Comment:
What about this?

```suggestion
                lines.append(f'* {description} in [#{pr_num}]({pr_link})')
```

This isn't *textually* the same as the release notes, but it's *effectively* the same, because of the auto-linking that GitHub does. It'll look a little worse as plain text (a lot worse than just the number), but nicer in rendered Markdown (also compared to just the number).

Reply:
The developer experience benefits (easily sync release notes and changelog when editing after automatic generation) come from the content being textually the same. This does look nice though, and we could get textual equality by making the same change to the release notes. Since we'd be making a change to the generated release notes anyway, I'd be on board with using `()` instead of `in` as well, so the line would this in both cases:
```py
                lines.append(f'* {description} ([#{pr_num}]({pr_link}))')
```